index.js

```diff
@@ -52,4 +52,261 @@ const state = {
+  productImage.setAttribute(`alt`, `${state.items[i].name}`)
+  produceList.append(storeItem)
+  storeItem.append(productImage)
+  produceList.append(addToCartButton)
+  document.querySelector('.item-list.store--item-list').append(produceList)
+}
+
+// - From the store, a user can add an item to their cart
+
+const addToCartButton = document.querySelectorAll('button')
+const unorderedList = document.querySelector('.item-list.cart--item-list')
+for (let i = 0; i < addToCartButton.length; i++){
+  let quantity = 0
+  addToCartButton[i].addEventListener('click', () => {
+      
+      
+    // - If the item is already in the cart, increase the item's quantity in the cart
+    quantity++
+      
+    function stateUpdate(){
+      state.cart.push({
+        id: state.items[i].id,
+        name: state.items[i].name,
+        price: 0.35,
+        quantity: quantity
+      })
+    }
+    stateUpdate()
+      
+    const productIcon = document.createElement('img')
+    productIcon.setAttribute("class", "cart--item-icon")
+    const productName = document.createElement('p')
+    const removeQuantityButton = document.createElement('button')
+    removeQuantityButton.setAttribute("class", "quantity-btn remove-btn center")
+    removeQuantityButton.innerText = '-'
+    const addQuantityButton = document.createElement('button')
+    addQuantityButton.setAttribute("class", "quantity-btn add-btn center")
+    addQuantityButton.innerText = '+'
+    const itemQuantity = document.createElement('span')
+    itemQuantity.setAttribute("class", "quantity-text center")
+    itemQuantity.innerText = quantity
+      
+    
+
+      
+      
+    let name = []
+    let indicies = []
+    for (let j = 0; j < state.cart.length; j++){
+  
+      name.push(state.cart[j].id)
+      let idx = name.indexOf(state.cart[j].id)
+      indicies.push(idx)
+      productIcon.setAttribute(`src`, `assets/icons/${state.cart[j].id}.svg`)
+      productIcon.setAttribute(`alt`, `${state.cart[j].name}`)
+      productName.innerText = `${state.cart[j].name}`
+    }
+      
+    const toFindRepeat = indicies => indicies.filter((item, index) => indicies.indexOf(item) !== index)
+    const duplicateElements = toFindRepeat(indicies)
+    let replaceIndex = []
+    for (let j = 0; j < indicies.length; j++){
+      const repeated = indicies.indexOf(duplicateElements[0], j)
+      replaceIndex.push(repeated)
+    }
+      
+      
+    if (duplicateElements.length === 1){
+      state.cart.splice(duplicateElements[0], 1, state.cart[replaceIndex.pop()])
+      state.cart.splice(replaceIndex.pop(), 1)
+    }
+      
+  
+    
+    if (quantity === 1){
+      const singleListItem = document.createElement('li')
+      unorderedList.append(singleListItem)
+      singleListItem.append(productIcon)
+      singleListItem.append(productName)
+      singleListItem.append(removeQuantityButton)
+      singleListItem.append(itemQuantity)
+      singleListItem.append(addQuantityButton)
```

Comment:
I you want to avoid repetition of these lines, you can add a comma after 'productIcon' and add the other elements you're appending.